packages/type-safe-api/scripts/generators/java-cdk-infrastructure/templates/mockIntegrations.mustache

```diff
@@ -0,0 +1,45 @@
+package {{#apiInfo}}{{#apis.0}}{{vendorExtensions.x-infrastructure-package}}{{/apis.0}}{{/apiInfo}};
+
+import {{#apiInfo}}{{#apis.0}}{{vendorExtensions.x-runtime-package}}{{/apis.0}}{{/apiInfo}}.JSON;
+import {{#apiInfo}}{{#apis.0}}{{vendorExtensions.x-runtime-package}}{{/apis.0}}{{/apiInfo}}.model.*;
+import software.aws.awsprototypingsdk.typesafeapi.Integrations;
+import software.aws.awsprototypingsdk.typesafeapi.MockIntegration;
+import software.aws.awsprototypingsdk.typesafeapi.MockIntegrationResponse;
+
+/**
+ * Type-safe mock integrations for API operations
+ */
+public class MockIntegrations {
+    static {
+        // Instantiate json instance so that any .toJson() methods can be used
+        new JSON();
```

Comment:
This is odd. Why do we create the instance without doing anything with it? Is there some janky singleton behaviour going on under the covers or something?

Reply:
Yeah exactly that - it's super janky haha!

The `.toJson` method in a model calls the static `JSON.getGson()` method ([see here](https://github.com/OpenAPITools/openapi-generator/blob/7881482161d22c138792cb9333daff216d6d68db/modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/pojo.mustache#L657)).

But the `gson` instance that `JSON.getGson()` returns is only initialised in a non-static initialiser block - it's only the act of creating some instance of `JSON` somewhere that initialises the `gson` instance:

https://github.com/OpenAPITools/openapi-generator/blob/7881482161d22c138792cb9333daff216d6d68db/modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/JSON.mustache#L112-L144

Reply:
That is so dumb - they should change that to a static initializer so it gets called the first time the class is loaded by the classloader...

Reply:
Yeah! No idea why it's not static... It caught me out when I built the lambda handler wrappers - things only failed at runtime which was super fun to debug 😆